tools/clang/lib/SPIRV/InitListHandler.cpp

```diff
@@ -385,6 +385,8 @@ InitListHandler::createInitForStructType(QualType type, SourceLocation srcLoc,
   llvm::SmallVector<SpirvInstruction *, 4> fields;
   const RecordDecl *structDecl = type->getAsStructureType()->getDecl();
   for (const auto *field : structDecl->fields()) {
+    if (field->getType()->isEmptyStructureType())
+      continue;
```

Comment:
this line actually causes the test to pass, but generates invalid SPIR-V:
```
%23 = OpCompositeConstruct %B %22
```

If I comment this out, and change the lines above, it fixes it.
```diff
--- a/tools/clang/lib/SPIRV/InitListHandler.cpp
+++ b/tools/clang/lib/SPIRV/InitListHandler.cpp
@@ -366,13 +366,15 @@ InitListHandler::createInitForStructType(QualType type, SourceLocation srcLoc,
     while (tryToSplitConstantArray())
       ;

-    auto init = initializers.back();
-    // We can only avoid decomposing and reconstructing when the type is
-    // exactly the same.
-    if (type.getCanonicalType() ==
-        init->getAstResultType().getCanonicalType()) {
-      initializers.pop_back();
-      return init;
+    if (initializers.size() > 0) {
+      auto init = initializers.back();
+      // We can only avoid decomposing and reconstructing when the type is
+      // exactly the same.
+      if (type.getCanonicalType() ==
+          init->getAstResultType().getCanonicalType()) {
+        initializers.pop_back();
+        return init;
+      }
     }

     // Otherwise, if the next initializer is a struct, it is not of the same

```

Reply:
Thanks, that seems to work better and I updated the PR.